*** src2.3.3/frameworks/base/include/camera/ICamera.h	2011-04-04 20:18:36.718480237 +0400
--- src_mock3.0.1/frameworks/base/include/camera/ICamera.h	2012-01-15 20:50:30.000000000 +0400
***************
*** 48,53 ****
--- 48,56 ----
      // pass the buffered ISurface to the camera service
      virtual status_t        setPreviewDisplay(const sp<ISurface>& surface) = 0;
  
+     // pass the preview texture. This is for 3.0 and higher versions of Android
+     setPreviewTexture(const sp<ISurfaceTexture>& surfaceTexture) = 0;
+ 
      // set the preview callback flag to affect how the received frames from
      // preview are handled.
      virtual void            setPreviewCallbackFlag(int flag) = 0;
