diff --git a/3rdparty/libpng/pngread.c b/3rdparty/libpng/pngread.c
index 24199689a7..0973248942 100644
--- a/3rdparty/libpng/pngread.c
+++ b/3rdparty/libpng/pngread.c
@@ -4111,7 +4111,7 @@ png_image_finish_read(png_imagep image, png_const_colorp background,
             row_stride = (png_int_32)/*SAFE*/png_row_stride;
 
          if (row_stride < 0)
-            check = -(png_uint_32)row_stride;
+            check = (png_uint_32)-row_stride;
 
          else
             check = (png_uint_32)row_stride;
diff --git a/3rdparty/libpng/pngwrite.c b/3rdparty/libpng/pngwrite.c
index 2c95eb66ca..0a974b8103 100644
--- a/3rdparty/libpng/pngwrite.c
+++ b/3rdparty/libpng/pngwrite.c
@@ -2030,7 +2030,7 @@ png_image_write_main(png_voidp argument)
             display->row_stride = (png_int_32)/*SAFE*/png_row_stride;
 
          if (display->row_stride < 0)
-            check = -(png_uint_32)display->row_stride;
+            check = (png_uint_32)-display->row_stride;
 
          else
             check = (png_uint_32)display->row_stride;
